elasticsearch/run.sh

```diff
@@ -57,17 +66,76 @@ else
 	exit 1
 fi
 
-TIMES=300
-function waitForES() {
-  for (( i=1; i<=$TIMES; i++ )); do
+# Wait for Elasticsearch port to be opened. Fail on timeout or if response from Elasticsearch is unexpected.
+wait_for_port_open() {
+	rm -f $LOG_FILE
     # test for ES to be up first and that our SG index has been created
-		result=$(curl --cacert $secret_dir/admin-ca --cert $secret_dir/admin-cert --key $secret_dir/admin-key -s -w "%{http_code}" -XGET "https://localhost:9200/.searchguard.${HOSTNAME}" -o /dev/null) ||:
-    [[ $result -eq 200 ]] && return 0
-    sleep 1
-  done
+	echo -n "Checking if Elasticsearch is ready on $ES_REST_BASEURL "
+	while ! response_code=$(curl -s \
```

Comment:
Not sure, but I'm assuming if the port is not opened, there will be no response_code, because there won't be an actual response, because the request didn't go through, so curl will exit with a non-zero code, which will cause the first part of the `||` condition to fail.  So it will sleep and retry.